app/routes/_layout+/kits.$kitId.release-custody.tsx

```diff
@@ -0,0 +1,180 @@
+import { json, redirect } from "@remix-run/node";
+import type { ActionFunctionArgs, LoaderFunctionArgs } from "@remix-run/node";
+
+import { Form, useLoaderData, useNavigation } from "@remix-run/react";
+import { z } from "zod";
+import { UserXIcon } from "~/components/icons/library";
+import { Button } from "~/components/shared/button";
+import { db } from "~/database/db.server";
+import { releaseCustody } from "~/modules/kit/service.server";
+import styles from "~/styles/layout/custom-modal.css?url";
+import { sendNotification } from "~/utils/emitter/send-notification.server";
+import { makeShelfError, ShelfError } from "~/utils/error";
+import { isFormProcessing } from "~/utils/form";
+import { data, error, getParams, parseData } from "~/utils/http.server";
+import {
+  PermissionAction,
+  PermissionEntity,
+} from "~/utils/permissions/permission.validator.server";
+import { requirePermission } from "~/utils/roles.server";
+
+export async function loader({ context, request, params }: LoaderFunctionArgs) {
+  const authSession = context.getSession();
+  const { userId } = authSession;
+
+  const { kitId } = getParams(params, z.object({ kitId: z.string() }), {
+    additionalData: { userId },
+  });
+
+  try {
+    await requirePermission({
+      userId,
+      request,
+      entity: PermissionEntity.kit,
+      action: PermissionAction.update,
+    });
+
+    const custody = await db.kitCustody
```

Comment:
FOr this one and the one above, i prefer to create service functions that we can just call in the loader to keep it clean. Inside the functions handle the errors. So in the loader we just see `getKitCustody(...)` or `getKit(...)`